src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -176,20 +175,27 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            DateTime now = DateTime.Now;
-            await InvokeJobFunction(now, false);
-
             // restart the timer with the next schedule occurrence
-            now = DateTime.Now;
-            DateTime nextOccurrence = _schedule.GetNextOccurrence(now);
-            TimeSpan nextInterval = nextOccurrence - now;
+            await InvokeJobFunction(DateTime.Now, false);
+            TimeSpan nextInterval = ScheduleStatus.Next - DateTime.Now;
```

Comment:
I see, yes, you're right. I misread the code above what you've changed and missed the line where `now` was being updated again. I agree that this was confusing, so a good change 👍